protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -471,6 +471,86 @@ func (rpccs *RPCConsumerServer) CancelSubscriptionContext(subscriptionKey string
 	}
 }
 
+func (rpccs *RPCConsumerServer) getEarliestBlockHashRequestedFromCacheReply(cacheReply *pairingtypes.CacheRelayReply) (int64, int64) {
+	blocksHashesToHeights := cacheReply.GetBlocksHashesToHeights()
+	earliestRequestedBlock := spectypes.NOT_APPLICABLE
+	latestRequestedBlock := spectypes.NOT_APPLICABLE
+
+	for _, blockHashToHeight := range blocksHashesToHeights {
+		if blockHashToHeight.Height >= 0 && (earliestRequestedBlock == spectypes.NOT_APPLICABLE || blockHashToHeight.Height < earliestRequestedBlock) {
+			earliestRequestedBlock = blockHashToHeight.Height
+		}
+		if blockHashToHeight.Height >= 0 && (latestRequestedBlock == spectypes.NOT_APPLICABLE || blockHashToHeight.Height > latestRequestedBlock) {
+			latestRequestedBlock = blockHashToHeight.Height
+		}
+	}
+	return latestRequestedBlock, earliestRequestedBlock
+}
+
+func (rpccs *RPCConsumerServer) resolveRequestedBlockAndUpdateExtensionIfNeeded(reqBlock int64, seenBlock int64, latestBlockHashRequested, earliestBlockHashRequested int64, addon string, protocolMessage chainlib.ProtocolMessage) int64 {
+	if reqBlock == spectypes.LATEST_BLOCK && seenBlock != 0 {
+		// make optimizer select a provider that is likely to have the latest seen block
+		reqBlock = seenBlock
+	}
+	// If latestBlockHashRequested provides more info on the relay we can set it as reqBlock
+	if latestBlockHashRequested >= 0 && (reqBlock == spectypes.LATEST_BLOCK || reqBlock < latestBlockHashRequested) {
```

Comment:
fix to overwrite only if values are defaultParsed, we can't overwrite other things that specifically parsed latestBlock